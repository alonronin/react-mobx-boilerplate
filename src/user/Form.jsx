import React from 'react';
import { observer } from 'mobx-react';

const Form = ({ store: { first, last, fullName, setState } = {} } = {}) => <form
	noValidate onSubmit={e => e.preventDefault()}
>
	<label>First Name <input
		type="text" value={first} onChange={e => setState({ first: e.target.value })}
	/></label>
	<label>Last Name <input
		type="text" value={last} onChange={e => setState({ last: e.target.value })}
	/></label>

	<p>{fullName}</p>
</form>;

export default observer(Form);
