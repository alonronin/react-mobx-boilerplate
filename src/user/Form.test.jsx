import React from 'react';
import Form from './Form';
import renderer from 'react-test-renderer';

describe('User', () => {
	test('<Form />', () => {
		const tree = renderer
			.create(<Form />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('<Form /> with store', () => {
		const store = {
			first: 'Test',
			last: 'Test',
			fullName: 'Test Test'
		};

		const tree = renderer
			.create(<Form store={store} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	})
});
