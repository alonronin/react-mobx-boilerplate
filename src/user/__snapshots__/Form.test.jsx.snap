// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User <Form /> 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <label>
    First Name 
    <input
      onChange={[Function]}
      type="text"
    />
  </label>
  <label>
    Last Name 
    <input
      onChange={[Function]}
      type="text"
    />
  </label>
  <p />
</form>
`;

exports[`User <Form /> with store 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <label>
    First Name 
    <input
      onChange={[Function]}
      type="text"
      value="Test"
    />
  </label>
  <label>
    Last Name 
    <input
      onChange={[Function]}
      type="text"
      value="Test"
    />
  </label>
  <p>
    Test Test
  </p>
</form>
`;
